t/t3438-rebase-git-logs-symlink.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
+
+test_description='git rebase when .git/logs is a symlink
+
+This test runs a "git rebase" in a repository where .git/logs
+is a symlink and checks that the symlink is not removed.
+If the system does not support symlinks, the whole test will
+be skipped. On Windows, the symlinks may be enabled by granting
+the "Create Symbolic Links" user right and settting the environment
+variable MSYS to "winsymlinks:nativestrict" in Git Bash.'
+
+. ./test-lib.sh
+
+if ! test_have_prereq SYMLINKS
+	then
+	    skip_all='Symlinks support is required for this test'
+	    test_done
+	fi
+
+# Create a rebase situation with 2 commits on a branch and another commit which
+# diverges from them. The 2 commits are created on a branch called
+# "rebase-reference" so that the rebase command that will be used later does
+# not depend on the name of the default branch (master, main, etc.).
+test_expect_success 'Create commits to prepare rebase' '
+    git switch --create rebase-reference &&
+    git commit --allow-empty -m "Initial commit" &&
+    git commit --allow-empty -m "Second commit" &&
+    git switch --create branch-to-rebase HEAD~1 &&
+    git commit --allow-empty -m "Dummy commit to rebase"
+'
+
+test_expect_success 'Turn .git/logs into a symlink' '
+    mv .git/logs actual_logs &&
+    ln -s ../actual_logs .git/logs
```

Comment:
As you pointed out in the comment above, `ln -s` won't work by default in an MSYS setup. However, that is exactly what all of the CI/PR builds run.

As a consequence, I fear that this test script will only ever run on Linux/macOS/FreeBSD, where we _know_ that things work.

A better idea might be to add a new test case to an existing `t34*` script, guarded by the `MINGW` prereq, that first verifies that it _can_ create a symbolic link, and that does not use `ln -s` for it, but `mklink /s` to do so, and if  it cannot create the symlink, simply renames the `logs` directory back, otherwise runs through the simple `rebase` and verifies that the symbolic link still exists.

Reply:
I decided to create the `SYMLINKS_WINDOWS` prerequisite based on the `SYMLINKS` prerequisite which uses `mklink` instead of `ln` in case it is useful to someone else in the future.